### DISCUSSION TITLE
[Questions] Onload and slow problem on biggest page with GET (Data provider)

### DISCUSSION THREAD

**Discussion by Maxiwork** [Budibase/budibase#7297] [Help] [answered]
![image](https://user-images.githubusercontent.com/111142619/184838452-1ecdffac-85b0-456a-87e6-c776a469305e.png)
So I have my GET on this page, i'm using the "On screen load"
This is "perfect" and I guess it saves time.

My question is : How can I get the data from the GET I did earlier on the component tab ?

![image](https://user-images.githubusercontent.com/111142619/184839430-9b823eb4-1eff-47ec-a79b-1a020debb87a.png)

Also, one more question : 

I have a page where I have multiple Data Provider / Repeater (GET) and that make the page so slow... Any idea why and how I can solve this problem ? I can't imagine my customers working on this page right now ^^"

**Comment by melohagan** [comment, marked as answer]
Hi @Maxiwork 

Thank you for raising this issue. The dev team is aware of the problem and working on a fix! https://github.com/Budibase/budibase/issues/7338

**Reply by melohagan** [reply]
As for multiple repeaters causing slow downs, you can always split out your functionality between screens if you find too much is going on within a single page.
